src/compiler/scala/tools/nsc/ast/Trees.scala

```diff
@@ -157,6 +169,12 @@ trait Trees extends scala.reflect.internal.Trees { self: Global =>
   }
 
   type ApiTransformer = super.Transformer
+
+  @nowarn("""cat=deprecation&origin=scala\.tools\.nsc\.ast\.Trees\.Transformer""")
+  final type AstTransformer = Transformer
+
+  @nowarn("msg=shadowing a nested class of a parent is deprecated")
+  @deprecated("use AstTransformer instead", since = "2.13.4")
   class Transformer extends InternalTransformer {
```

Comment:
Please don't deprecate `Transformer`. It's what every compiler plugin in existence extends. If it's deprecated, there's no way to cross-compile a compiler plugin without warning anymore.

Reply:
how would you feel about deprecating/renaming the base member for this and `Cursor` instead? from the perspective of removing inner class shadowing, I don't think it makes any particular difference

Reply:
another possibility I suppose would be to introduce the replacement types without deprecating the current types, and then deprecate the current types later, to improve cross-compiling

Reply:
That works too.

Reply:
addressed in 842e46d652539d857222e0d677fb5ecdc3bee8ba